packages/node-runtime-worker-thread/src/child-process-mongosh-bus.ts

```diff
@@ -0,0 +1,23 @@
+import { ChildProcess } from 'child_process';
+import { MongoshBus } from '@mongosh/types';
+import { exposeAll, WithClose } from './rpc';
+
+export class ChildProcessMongoshBus {
+  exposedEmitter: WithClose<MongoshBus>;
+
+  constructor(eventEmitter: MongoshBus, childProcess: ChildProcess) {
+    const exposedEmitter: WithClose<MongoshBus> = exposeAll(
+      {
+        emit(...args) {
+          eventEmitter.emit(...args);
+        },
+        on() {
+          // no-op
+          return exposedEmitter;
```

Comment:
Is this being called somewhere? If not, could we maybe throw an exception instead, so that we don’t accidentally start using this inside of shell code?

Reply:
Yeah, it's not used anywhere and yes, you're right, throwing something meaningful when called is a good idea, will add!

Reply:
Done in ad8136d